From 55ad273328f2028b71d39f552108938082ab831a Mon Sep 17 00:00:00 2001
From: MTR <github@hilscher.com>
Date: Tue, 17 Mar 2020 10:43:48 +0100
Subject: [PATCH] Revert "MLK-21756 brcmfmac: bcmsdh: attach device only func1
 compatible string match"

This commit introduced a check of a compatible string which does not exist in a sdio_func[] entry,
thus the probe will always fail on "default" bcm4329 usage

This reverts commit 8660095dde5dffe4dd0a781e8f8fb5ee00237a3d.
---
 drivers/net/wireless/broadcom/brcm80211/brcmfmac/bcmsdh.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/bcmsdh.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/bcmsdh.c
index 6ca0f8b87af3..0380ae9caa63 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/bcmsdh.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/bcmsdh.c
@@ -33,7 +33,6 @@
 #include <linux/errno.h>
 #include <linux/module.h>
 #include <linux/acpi.h>
-#include <linux/of.h>
 #include <net/cfg80211.h>
 
 #include <defs.h>
@@ -1009,7 +1008,6 @@ static int brcmf_ops_sdio_probe(struct sdio_func *func,
 	struct brcmf_sdio_dev *sdiodev;
 	struct brcmf_bus *bus_if;
 	struct device *dev;
-	struct device *func_dev;
 
 	brcmf_dbg(SDIO, "Enter\n");
 	brcmf_dbg(SDIO, "Class=%x\n", func->class);
@@ -1025,11 +1023,6 @@ static int brcmf_ops_sdio_probe(struct sdio_func *func,
 	/* prohibit ACPI power management for this device */
 	brcmf_sdiod_acpi_set_power_manageable(dev, 0);
 
-	func_dev = &func->card->sdio_func[0]->dev;
-	if (!func_dev->of_node ||
-	    !of_device_is_compatible(func_dev->of_node, "brcm,bcm4329-fmac"))
-		return -ENODEV;
-
 	/* Consume func num 1 but dont do anything with it. */
 	if (func->num == 1)
 		return 0;
-- 
2.17.1

